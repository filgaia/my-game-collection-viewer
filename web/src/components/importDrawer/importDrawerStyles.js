const importDrawerStyles = (theme) => ({
    listItem: {
        marginRight: 0
    },
    input: {
        display: 'none'
    },
    labelFor: {
        cursor: 'pointer'
    }
});

export default importDrawerStyles;
