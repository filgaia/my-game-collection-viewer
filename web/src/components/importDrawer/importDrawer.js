// @vendors
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
// @material
import {
  Drawer,
  IconButton,
  List,
  ListItem,
  ListItemIcon,
  Tooltip,
} from '@mui/material';
// https://material.io/tools/icons/?style=baseline
import {
  CloudDownload,
} from '@mui/icons-material';
import withStyles from '@mui/styles/withStyles';
// @styles
import importDrawerStyles from './importDrawerStyles';
// @constants
import { IMPORT_DRAWER } from '../../constants/index';
// @components
import Login from '../login/login';

export class ImportDrawer extends Component {
  static buildLogoutButton() {
    return <Login />;
  }

  constructor(props) {
    super(props);

    this.onImportFile = this.onImportFile.bind(this);
    this.onToggleDrawer = this.onToggleDrawer.bind(this);
  }

  onImportFile(event) {
    const { importFile } = this.props;
    importFile(get(event, 'target.files.0'));
  }

  onToggleDrawer(open) {
    const { toggleDrawer } = this.props;
    const response = { drawer: IMPORT_DRAWER, open };
    toggleDrawer({ response });
  }

  buildItems() {
    const { classes } = this.props;
    const logoutButton = ImportDrawer.buildLogoutButton();
    const options = [
      { icon: ['fab', 'playstation'], title: 'Coming Soon...' },
      { icon: ['fab', 'xbox'], title: 'Coming Soon...' },
    ];

    const items = options.map((item, index) => (
      <ListItem button key={`Ã¬mport-${index}`}>
        <ListItemIcon className={classes.listItem}>
          <Tooltip title={item.title} aria-label={item.title}>
            <FontAwesomeIcon icon={item.icon} size="lg" />
          </Tooltip>
        </ListItemIcon>
      </ListItem>
    ));

    return (
      <div>
        <List>
          <ListItem button key="import-file">
            <label htmlFor="import-file" className={classes.labelFor}>
              <input accept=".c, application/x-zip-compressed" className={classes.input} id="import-file" type="file" onChange={this.onImportFile} />
              <ListItemIcon className={classes.listItem}>
                <Tooltip title="From file" aria-label="From file">
                  <FontAwesomeIcon icon="file-import" size="lg" />
                </Tooltip>
              </ListItemIcon>
            </label>
          </ListItem>
          <ListItem button key="import-drive">
            {logoutButton}
          </ListItem>
          {items}
        </List>
      </div>
    );
  }

  render() {
    const { importDrawer } = this.props;
    const items = this.buildItems();

    return (
      <>
        <IconButton color="inherit" onClick={() => this.onToggleDrawer(true)} size="large">
          <Tooltip title="Add Games..." aria-label="Add Games...">
            <CloudDownload />
          </Tooltip>
        </IconButton>
        <Drawer anchor="right" open={importDrawer} onClose={() => this.onToggleDrawer(false)}>
          {items}
        </Drawer>
      </>
    );
  }
}

ImportDrawer.propTypes = {
  classes: PropTypes.object.isRequired,
  importDrawer: PropTypes.bool.isRequired,
  importFile: PropTypes.func.isRequired,
  toggleDrawer: PropTypes.func.isRequired,
};

export default withStyles(importDrawerStyles)(ImportDrawer);
