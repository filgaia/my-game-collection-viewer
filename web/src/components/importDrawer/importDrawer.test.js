import React from 'react';
import renderer from 'react-test-renderer';
import { ImportDrawer } from './importDrawer';

describe('<ImportDrawer />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ImportDrawer
        classes={{}}
        importDrawer={false}
        importFile={jest.fn()}
        toggleDrawer={jest.fn()}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
