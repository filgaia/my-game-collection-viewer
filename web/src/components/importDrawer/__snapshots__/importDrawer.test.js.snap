// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImportDrawer /> renders correctly 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <svg
    aria-hidden={true}
    aria-label="Add Games..."
    aria-labelledby={null}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-mui-internal-clone-element={true}
    data-testid="CloudDownloadIcon"
    focusable="false"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
    />
  </svg>
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
